### DISCUSSION TITLE
Hyper-V instance running dead slow on WebDAV import / sync

### DISCUSSION THREAD

**Discussion by NaXal** [photoprism/photoprism#2692] [General]
Hello,

I have PhotoPrism running on a Hyper-V VM with Ubuntu 22 using docker. Just as the installation tutorial suggests with the YML file. 

Previously I have used same exact method with VMWare as hypervisor on the same system and network.

VMWare based operation was somewhat fine and relatively trouble free however, with Hyper-V VM, WebDAV sync / import from nextcloud is dead slow. Firstly, the importing doesn't start immediately. It may start an hour later may be. Even when trying to manually trigger the import. But the biggest issue is the dead slow importing process. One image every 30/40 second may be?

Following are my configuration.

Host System :- Core i7 4790 / 32GB DDR3 / 500 GB SSD (VM Drive)
Host OS :- Windows 11 Pro
Hypervisor :- Hyper-V Generation 2 VM (8 Cores / 16GB RAM / External vSwitch) / Ubuntu 22 LTS
PhotoPrism :- with Docker / Compose

Data is pulled from another system running in the network (Home LAN 1GBPS) with NextCloud snap installation. I have tested ping between two systems (NextCloud and PhotoPrism) and iPerf. Both are fine, ping is under 1ms as expected between LAN, so is iPerf, giving steady 950mbps+ throughput

Earlier, with VMWare, I could see heavy HDD and CPU/RAM usage while importing going on but here, Hyper-V instance is showing next to nothing on load and importing one picture at a time, that too like 30/40 seconds apart. 

Hyper-V Ubuntu system isn't that slow, if I remove PhotoPrism and install something else, it is able to use full system resource of the VM but with PhotoPrism, its running dead slow on importing process.

Any idea where I should start the troubleshooting?

Thanks.

**Comment by lastzero** [comment]
We have started working on a Windows-specific troubleshooting guide: https://docs.photoprism.app/getting-started/troubleshooting/windows/

Feel free to send a pull request for additional recommendations once this is solved. There's a chance the existing information points you in the right direction. In general, native filesystem performance will always be better than Docker Desktop. I've especially observed this on macOS. Windows shouldn't be that much slower though.

**Comment by NaXal** [comment]
Thank you for the reply. I went through the article.

Just to point, 

PhotoPrism isn't running via WSL (Subsystem) or Windows Docker Desktop, its a full VM on Hyper-V
I am not mounting the entire Physical Disk.
PhotoPrism is on a Hyper-V VM with Ubuntu on a Virtual Disk with ext4 partitioning scheme.

It is no different than any other hyper visor like VMWare or VirtualBox, just it's on Hyper-V.

Anyway, I shall wait for further documentation if or when available.

Thanks.

**Reply by lastzero** [reply]
We'll likely not be able to test that specific configuration anytime soon. On Windows, we only test with Docker. Desktop. Once we (maybe) have a native binary, we'll test that as well.

**Reply by NaXal** [reply]
Thank you for informing. I can understand your resource and time constraints. Just to be clear, Hyper-V is a hypervisor allowing an Windows user to create Virtual Machines, where Linux OS, like Ubuntu Server 22 can be installed along with Linux version of Docker / Docker-Compose to run containers.

Hyper-V is similar in function like Virtual Box or VMWare for creating virtual machines. 

I think Windows Binary is for native Windows based installation, not sure if that is somehow related to Hyper-V or not. I am no expert. Just an user.

Anyway, thank you for informing me outright. 

I would try for any alternative means of running PhotoPrism !!

Thanks.
